tests/sentry/utils/test_registry.py

```diff
@@ -33,3 +33,22 @@ def unregistered_func():
 
         test_registry.register("something else")(unregistered_func)
         assert test_registry.get("something else") == unregistered_func
+
+    def test_allow_duplicate_values(self):
+        test_registry = Registry[str](enable_reverse_lookup=False)
+
+        @test_registry.register("something")
+        @test_registry.register("something 2")
+        def registered_func():
+            pass
+
+        assert test_registry.get("something") == registered_func
```

Comment:
Mind adding an assertion to ensure both keys are registered?
```suggestion
        assert test_registry.get("something") == registered_func
        assert test_registry.get("something 2") == registered_func
```